x-pack/plugins/aiops/public/components/log_categorization/sampling_menu/sampling_menu.tsx

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useCallback, useMemo, useState } from 'react';
+import {
+  EuiFlexItem,
+  EuiPopover,
+  EuiPanel,
+  EuiSpacer,
+  EuiCallOut,
+  EuiSelect,
+  EuiFormRow,
+  EuiButtonEmpty,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import useObservable from 'react-use/lib/useObservable';
+
+import { RandomSamplerRangeSlider } from './random_sampler_range_slider';
+import {
+  RandomSampler,
+  RandomSamplerOption,
+  RANDOM_SAMPLER_OPTION,
+  RANDOM_SAMPLER_SELECT_OPTIONS,
+} from './random_sampler';
+
+interface Props {
+  randomSampler: RandomSampler;
+  reload: () => void;
+}
+
+export const SamplingMenu: FC<Props> = ({ randomSampler, reload }) => {
+  const [showSamplingOptionsPopover, setShowSamplingOptionsPopover] = useState(false);
+
+  const samplingProbability = useObservable(
+    randomSampler.getProbability$(),
+    randomSampler.getProbability()
+  );
+  const setSamplingProbability = useCallback(
+    (probability: number | null) => {
+      randomSampler.setProbability(probability);
+      reload();
+    },
+    [reload, randomSampler]
+  );
+
+  const randomSamplerPreference = useObservable(randomSampler.getMode$(), randomSampler.getMode());
+  const setRandomSamplerPreference = useCallback(
```

Comment:
Currently the random sampler settings are getting reset every time the flyout is opened. This means I am having to reconfigure it every time I want to run a new analysis. Could / should the settings be remembered?

Reply:
Adding the sampling settings to local storage in https://github.com/elastic/kibana/pull/153449/commits/c3aaffaa49b94bb5cc7818e4690562370d4edcac